butterfree/pipelines/feature_set_pipeline_metadata.py

```diff
@@ -0,0 +1,220 @@
+from typing import List, Literal, Optional, Union
+
+from pydantic import BaseModel, ConfigDict, Field, field_serializer
+from typing_extensions import Annotated
+
+from butterfree.extract.readers.reader import (
+    FileReaderMetadata,
+    KafkaReaderMetadata,
+    TableReaderMetadata,
+)
+from butterfree.load.writers.writer import WriterMetadata
+from butterfree.pipelines import FeatureSetPipeline
+from butterfree.transform.aggregated_feature_set import AggregatedFeatureSet
+
+
+class Column(BaseModel):
+    """Metadata model for a column in a feature set.
+
+    This model represents the metadata of a single column in a feature set,
+    including its name, data type, and whether it's a primary key.
+    """
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    name: str = Field(..., description="The name of the column")
+    data_type: str = Field(
+        ...,
+        description=(
+            "The data type of the column (e.g., StringType, IntegerType) represented by pyspark.sql.types"  # noqa: E501
+        ),
+    )
+    description: str = Field(
+        ...,
+        description="The description of the column",
+    )
+    primary_key: bool = Field(
+        ...,
+        description="Whether the column is a primary (or partition if it's a Cassandra table) key",  # noqa: E501
+    )
+
+
+class Catalog(BaseModel):
+    """Metadata model for a feature set catalog.
+
+    This model represents the catalog information of a feature set,
+    including its name, description, and column definitions.
+    """
+
+    feature_set_name: str = Field(..., description="The name of the Feature Set")
```

Comment:
Maybe use this field to recommend a format?

Reply:
How so? Like a pattern? Characters separated by underscore?